### PR TITLE
feat(next): grant access token and check expiration

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Implement expiration policy:

1. set cookie's max-age to the same as refresh token
2. if the access token is valid, the session is authenticated
3. if the access token is expired, try to grant new with refresh token

## Testing

Local tested with sample code.

### CODE REVIEW

File: packages/next/src/index.ts [hunk 1]
```diff
@@ -7,6 +7,17 @@ import { LogtoNextConfig, LogtoUser } from './types';
 
 export type { LogtoUser } from './types';
 
+// Refresh token can be revoked, so it is authenticated only when we have a unexpired access token
+const checkIsAuthenticatedByAccessToken = async (client: NodeClient): Promise<boolean> => {
+  try {
+    await client.getAccessToken();
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/next/src/index.ts:13]: Refresh token can be revoked?

[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/next/src/index.ts:13]: LOG-3772 to refactor this.

File: packages/next/src/index.ts [hunk 2]
```diff
@@ -91,6 +103,7 @@ export default class LogtoClient {
       password: this.config.cookieSecret,
       cookieOptions: {
         secure: this.config.cookieSecure,
+        maxAge: 14 * 24 * 60 * 60,
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/next/src/index.ts:106]: created LOG-3773 to align

File: packages/next/src/storage.ts [hunk 3]
```diff
@@ -21,9 +23,15 @@ export default class NextStorage implements Storage {
...
+    this.sessionChanged = true;
   }
 
   async save() {
+    if (!this.sessionChanged) {
+      return;
+    }
+
     await this.request.session.save();
+    this.sessionChanged = false;
```
Comment (demonzoo) [inline packages/next/src/storage.ts:35]: What if the session changed to `true` when saving the session, in a high concurrency situation?

**Review by demonzoo — APPROVED** [review]

**Pull request merged**